lib/videoroom/meeting.ex

```diff
@@ -0,0 +1,120 @@
+defmodule Videoroom.Meeting do
+  @moduledoc """
+  A Module responsible for handling a room.
+  """
+  use GenServer
+
+  alias Jellyfish.ServerMessage.PeerCrashed
+  alias Jellyfish.ServerMessage.PeerDisconnected
+  alias Jellyfish.ServerMessage.RoomCrashed
+
+  alias Jellyfish.Room
+
+  @room_table :room_table
+
+  # Api
+
+  @spec start_link(Keyword.t()) :: GenServer.on_start()
+  def start_link(args) do
+    GenServer.start_link(__MODULE__, args,
+      name: {:via, Registry, {Videoroom.Registry, args[:name]}}
+    )
+  end
+
+  @spec add_peer(pid()) :: {:ok, binary()} | {:error, binary()}
+  def add_peer(meeting) do
+    GenServer.call(meeting, {:add_peer})
+  end
+
+  # Callbacks
+
+  @impl true
+  def init(args) do
+    client = Jellyfish.Client.new()
+
+    with {:ok, notifier} <- Jellyfish.Notifier.start(),
+         {:ok, room} <- find_or_create_room(client, args) do
+      Process.monitor(notifier)
+
+      {:ok, %{client: client, notifier: notifier, name: args[:name], room_id: room.id}}
+    else
+      {:error, reason} ->
+        raise("Failed to create a meeting, reason: #{inspect(reason)}")
+    end
+  end
+
+  defp find_or_create_room(client, opts) do
+    name = opts[:name]
```

Comment:
1. In the code above you're using `args`, not opts
2. I'd not pass a keyword here but rather explicit arguments, i.e.
```suggestion
  defp find_or_create_room(client, room_name, max_peers) do
```